src/models/split_transfers.py

```diff
@@ -44,20 +51,27 @@ def __init__(self, period: AccountingPeriod, mixed_transfers: list[Transfer]):
         self.cow_transfers: list[Transfer] = []
 
     def _process_native_transfers(
-        self, indexed_slippage: dict[Address, SolverSlippage], log_saver: PrintStore
+        self, indexed_slippage: dict[Address, SolverSlippage]
     ) -> int:
+        """
+        Draining the `unprocessed_native` (ETH) transfers into processed
+        versions as `eth_transfers`. Processing adjusts for negative slippage by deduction.
+        """
         penalty_total = 0
         while self.unprocessed_native:
             transfer = self.unprocessed_native.pop(0)
             solver = transfer.receiver
             slippage: Optional[SolverSlippage] = indexed_slippage.get(solver)
             if slippage is not None:
+                assert (
+                    slippage.amount_wei < 0
```

Comment:
Can the slippage as defined in line 64 above still only be non-positive?

Reply:
It is in fact only negative. This assertions isn't technically necessary because of the SplitSlippage structure definition. However since this class doesn't own or know anything about the slippage structure definition, without this assertion it would have to trust that it was receiving the negative component.

Reply:
Have also added analogous assertion to positives in https://github.com/cowprotocol/solver-rewards/pull/163/commits/4057c8b4dad10cc6a1ef312dcef570114772543f

Reply:
I think it would be nice if these "hidden assumptions" are somehow indicated by the function signature. E.g. you could rename: indexed_slippage to indexed_negative_slippage